veracode/api.py

```diff
@@ -24,14 +28,16 @@ class VeracodeXMLAPI:
     https://help.veracode.com/reader/LMv_dtSHyb7iIxAQznC~9w/pd_p6JjB9PcDNH3GzWF5Ag
     """
 
-    def __init__(self):
-        # Hard code these to None as they should be specified in the derived classes
-        self._app_id = None
+    def __init__(self, app_name: str):
         self._version = None
 
         ## Use the setter to apply a default to ensure it is valid
         self.base_url = "https://analysiscenter.veracode.com/api/"
 
+        # Set app name and look up ID
+        self._app_name = app_name
```

Comment:
Shouldn't we be using the setter to ensure it's valid?
```suggestion
        self.app_name = app_name
```